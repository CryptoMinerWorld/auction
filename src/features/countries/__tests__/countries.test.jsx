import React from 'react';
import {
  render, fireEvent, cleanup, wait,
} from 'react-testing-library';
import 'jest-dom/extend-expect';
import { MockedProvider } from 'react-apollo/test-utils';
import gql from 'graphql-tag';
import CountryPage from '../index';

jest.mock('firebase');

afterEach(cleanup);


const mocks = [
  {
    request: {
      query: gql`
        {
          user(id: "0xd9b74f73d933fde459766f74400971b29b90c9d2") {
            name
          }
        }
      `,
      variables: { id: '123' },
    },
    result: {
      data: {
        item: 'Josh',
      },
    },
  },
];

const handleBuyNow = jest.fn();

describe('Country Map', () => {
  test('map page renders', () => {
    const { getByTestId, getByText } = render(
      <MockedProvider mocks={mocks}>
        <CountryPage handleBuyNow={handleBuyNow} />
      </MockedProvider>,
    );
    expect(getByTestId('mapPage')).toBeInTheDocument();
    expect(getByTestId('mapComponent')).toBeInTheDocument();
    expect(getByTestId('cartComponent')).toBeInTheDocument();
    expect(getByText('John Brown')).toBeInTheDocument();
  });

  test.skip('map matches snapshot', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={mocks}>
        <CountryPage handleBuyNow={handleBuyNow} />
      </MockedProvider>,
    );
    expect(getByTestId('mapPage')).toMatchSnapshot();
  });

  test('when I hover on a country its details appear in the detail bar', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={mocks}>
        <CountryPage handleBuyNow={handleBuyNow} />
      </MockedProvider>,
    );
    fireEvent.mouseEnter(getByTestId('Brazil'));
    expect(getByTestId('countryDetails')).toHaveTextContent('Brazil');
  });

  test('when I select a countrty it gets added to the cart', () => {
    const { getByText, getByTestId } = render(
      <MockedProvider mocks={mocks}>
        <CountryPage handleBuyNow={handleBuyNow} />
      </MockedProvider>,
    );
    fireEvent.click(getByTestId('India'));
    expect(getByText('India')).toBeInTheDocument();
  });

  test('when I select a country in the cart it gets removed', () => {
    const { getByText, getByTestId, queryByText } = render(
      <MockedProvider mocks={mocks}>
        <CountryPage handleBuyNow={handleBuyNow} />
      </MockedProvider>,
    );
    fireEvent.click(getByTestId('India'));
    expect(getByText('India')).toBeInTheDocument();
    fireEvent.click(getByTestId('remove-India'));
    expect(queryByText('India')).not.toBeInTheDocument();
  });

  test('when I buy items the buy now metamask modal appears', () => {
    const { getByTestId, getByText, queryByTestId } = render(
      <MockedProvider mocks={mocks}>
        <CountryPage handleBuyNow={handleBuyNow} />
      </MockedProvider>,
    );
    fireEvent.click(getByTestId('India'));
    expect(getByText('India')).toBeInTheDocument();
    getByTestId('cartLoading');

    wait(() => expect(
      queryByTestId('cartLoading')
        .toBeNull(),
    ));
    getByTestId('buyNow');
    fireEvent.click(getByTestId('buyNow'));
    expect(handleBuyNow).toBeCalled();
    expect(handleBuyNow).toBeCalledWith([{
      country: 'Brazil', key: '1', plots: 44, price: 32, return: 54, roi: 45,
    }, {
      country: 'India', key: '1', plots: 44, price: 32, return: 54, roi: 45,
    }]);
  });

  test('once I have bought a country it appears in my profile', () => {
    expect(true).toBeFalsy();
  });
});


test.skip('info bar aggregates price when a country is not selected ', () => {
  expect(true).toBeFalsy();
});

test.skip('cancelling the metamsk prompt also cancels the buy now modal ', () => {
  expect(true).toBeFalsy();
});

test.skip('when I hover on a countrty its details appear in a tool tip', () => {
  expect(true).toBeFalsy();
});

test.skip('map details update in realtime', () => {
  expect(true).toBeFalsy();
});

test.skip('You shoud only be able to add a country once', () => {
  expect(true).toBeFalsy();
});

// tab structure on workshop
// profile country details update in realtime?
// gifting a country
// handling transactions
