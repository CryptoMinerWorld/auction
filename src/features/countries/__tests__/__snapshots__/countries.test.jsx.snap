// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Map map matches snapshot 1`] = `
<div
  data-testid="mapPage"
>
  <div
    class="flex"
  >
    <div
      class="w-third pa3"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-default ant-table-scroll-position-left"
            >
              <div
                class="ant-table-content"
              >
                <div
                  class="ant-table-body"
                >
                  <table
                    class=""
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          <div>
                            Name
                          </div>
                          <i
                            class="anticon anticon-smile-o ant-table-filter-icon ant-dropdown-trigger"
                            style="color: rgb(170, 170, 170);"
                            title="Filter menu"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="smile"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M288 421a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm352 0a48 48 0 1 0 96 0 48 48 0 1 0-96 0zM512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm263 711c-34.2 34.2-74 61-118.3 79.8C611 874.2 562.3 884 512 884c-50.3 0-99-9.8-144.8-29.2A370.4 370.4 0 0 1 248.9 775c-34.2-34.2-61-74-79.8-118.3C149.8 611 140 562.3 140 512s9.8-99 29.2-144.8A370.4 370.4 0 0 1 249 248.9c34.2-34.2 74-61 118.3-79.8C413 149.8 461.7 140 512 140c50.3 0 99 9.8 144.8 29.2A370.4 370.4 0 0 1 775.1 249c34.2 34.2 61 74 79.8 118.3C874.2 413 884 461.7 884 512s-9.8 99-29.2 144.8A368.89 368.89 0 0 1 775 775zM664 533h-48.1c-4.2 0-7.8 3.2-8.1 7.4C604 589.9 562.5 629 512 629s-92.1-39.1-95.8-88.6c-.3-4.2-3.9-7.4-8.1-7.4H360a8 8 0 0 0-8 8.4c4.4 84.3 74.5 151.6 160 151.6s155.6-67.3 160-151.6a8 8 0 0 0-8-8.4z"
                              />
                            </svg>
                          </i>
                        </th>
                        <th
                          class=""
                        >
                          <div>
                            Age
                          </div>
                        </th>
                        <th
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          <div>
                            Address
                          </div>
                          <i
                            class="anticon anticon-filter ant-dropdown-trigger"
                            title="Filter menu"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="filter"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                              />
                            </svg>
                          </i>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          <span
                            class="ant-table-row-indent indent-level-0"
                            style="padding-left: 0px;"
                          />
                          John Brown
                        </td>
                        <td
                          class=""
                        >
                          32
                        </td>
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          New York No. 1 Lake Park
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          <span
                            class="ant-table-row-indent indent-level-0"
                            style="padding-left: 0px;"
                          />
                          Joe Black
                        </td>
                        <td
                          class=""
                        >
                          42
                        </td>
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          London No. 1 Lake Park
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="3"
                      >
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          <span
                            class="ant-table-row-indent indent-level-0"
                            style="padding-left: 0px;"
                          />
                          Jim Green
                        </td>
                        <td
                          class=""
                        >
                          32
                        </td>
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          Sidney No. 1 Lake Park
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="4"
                      >
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          <span
                            class="ant-table-row-indent indent-level-0"
                            style="padding-left: 0px;"
                          />
                          Jim Red
                        </td>
                        <td
                          class=""
                        >
                          32
                        </td>
                        <td
                          class="ant-table-column-has-actions ant-table-column-has-filters"
                        >
                          London No. 2 Lake Park
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination"
              unselectable="unselectable"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-disabled ant-pagination-prev"
                title="Previous Page"
              >
                <a
                  class="ant-pagination-item-link"
                >
                  <i
                    class="anticon anticon-left"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="left"
                      fill="currentColor"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </i>
                </a>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a>
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-disabled ant-pagination-next"
                title="Next Page"
              >
                <a
                  class="ant-pagination-item-link"
                >
                  <i
                    class="anticon anticon-right"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="right"
                      fill="currentColor"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                      />
                    </svg>
                  </i>
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          style="position: absolute; top: 0px; left: 0px; width: 100%;"
        >
          <div>
            <div
              class="ant-dropdown  ant-dropdown-placement-bottomRight  ant-dropdown-hidden"
            >
              <div>
                <div
                  class="custom-filter-dropdown"
                >
                  <input
                    class="ant-input"
                    placeholder="Search name"
                    type="text"
                    value=""
                  />
                  <button
                    class="ant-btn ant-btn-primary"
                    type="button"
                  >
                    <span>
                      Search
                    </span>
                  </button>
                  <button
                    class="ant-btn"
                    type="button"
                  >
                    <span>
                      Reset
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="position: absolute; top: 0px; left: 0px; width: 100%;"
        >
          <div>
            <div
              class="ant-dropdown  ant-dropdown-placement-bottomRight  ant-dropdown-hidden"
            >
              <div
                class="ant-table-filter-dropdown"
              >
                <ul
                  class="ant-dropdown-menu ant-dropdown-menu-without-submenu ant-dropdown-menu-root ant-dropdown-menu-vertical"
                  role="menu"
                  tabindex="0"
                >
                  <li
                    class="ant-dropdown-menu-item"
                    role="menuitem"
                  >
                    <label
                      class="ant-checkbox-wrapper"
                    >
                      <span
                        class="ant-checkbox"
                      >
                        <input
                          class="ant-checkbox-input"
                          type="checkbox"
                          value=""
                        />
                        <span
                          class="ant-checkbox-inner"
                        />
                      </span>
                    </label>
                    <span>
                      London
                    </span>
                  </li>
                  <li
                    class="ant-dropdown-menu-item"
                    role="menuitem"
                  >
                    <label
                      class="ant-checkbox-wrapper"
                    >
                      <span
                        class="ant-checkbox"
                      >
                        <input
                          class="ant-checkbox-input"
                          type="checkbox"
                          value=""
                        />
                        <span
                          class="ant-checkbox-inner"
                        />
                      </span>
                    </label>
                    <span>
                      New York
                    </span>
                  </li>
                </ul>
                <div
                  class="ant-table-filter-dropdown-btns"
                >
                  <a
                    class="ant-table-filter-dropdown-link confirm"
                  >
                    OK
                  </a>
                  <a
                    class="ant-table-filter-dropdown-link clear"
                  >
                    Reset
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-two-thirds pa3"
    />
  </div>
  <div
    data-testid="cartComponent"
  >
    <article
      class="pa3 pt4 tr bg-dark-gray white"
      data-name="slab-stat"
    >
      <dl
        class="dib mr5"
      >
        <dd
          class="f6 f5-ns b ml0"
        >
          Closed Issues
        </dd>
        <dd
          class="f3 f2-ns b ml0"
        >
          1,024
        </dd>
      </dl>
      <dl
        class="dib mr5"
      >
        <dd
          class="f6 f5-ns b ml0"
        >
          Open Issues
        </dd>
        <dd
          class="f3 f2-ns b ml0"
        >
          993
        </dd>
      </dl>
      <dl
        class="dib mr5"
      >
        <dd
          class="f6 f5-ns b ml0"
        >
          Next Release
        </dd>
        <dd
          class="f3 f2-ns b ml0"
        >
          10-22
        </dd>
      </dl>
    </article>
    <div
      class="flex"
    >
      <div
        class="w-third"
      >
        <p>
          Loading...
        </p>
      </div>
      <div
        class="w-two-thirds"
      >
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-default ant-table-scroll-position-left"
              >
                <div
                  class="ant-table-content"
                >
                  <div
                    class="ant-table-body"
                  >
                    <table
                      class=""
                    >
                      <colgroup>
                        <col />
                        <col />
                        <col />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <div>
                              Name
                            </div>
                            <i
                              class="anticon anticon-smile-o ant-table-filter-icon ant-dropdown-trigger"
                              style="color: rgb(170, 170, 170);"
                              title="Filter menu"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="smile"
                                fill="currentColor"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M288 421a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm352 0a48 48 0 1 0 96 0 48 48 0 1 0-96 0zM512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm263 711c-34.2 34.2-74 61-118.3 79.8C611 874.2 562.3 884 512 884c-50.3 0-99-9.8-144.8-29.2A370.4 370.4 0 0 1 248.9 775c-34.2-34.2-61-74-79.8-118.3C149.8 611 140 562.3 140 512s9.8-99 29.2-144.8A370.4 370.4 0 0 1 249 248.9c34.2-34.2 74-61 118.3-79.8C413 149.8 461.7 140 512 140c50.3 0 99 9.8 144.8 29.2A370.4 370.4 0 0 1 775.1 249c34.2 34.2 61 74 79.8 118.3C874.2 413 884 461.7 884 512s-9.8 99-29.2 144.8A368.89 368.89 0 0 1 775 775zM664 533h-48.1c-4.2 0-7.8 3.2-8.1 7.4C604 589.9 562.5 629 512 629s-92.1-39.1-95.8-88.6c-.3-4.2-3.9-7.4-8.1-7.4H360a8 8 0 0 0-8 8.4c4.4 84.3 74.5 151.6 160 151.6s155.6-67.3 160-151.6a8 8 0 0 0-8-8.4z"
                                />
                              </svg>
                            </i>
                          </th>
                          <th
                            class=""
                          >
                            <div>
                              Age
                            </div>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <div>
                              Address
                            </div>
                            <i
                              class="anticon anticon-filter ant-dropdown-trigger"
                              title="Filter menu"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="filter"
                                fill="currentColor"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                />
                              </svg>
                            </i>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <span
                              class="ant-table-row-indent indent-level-0"
                              style="padding-left: 0px;"
                            />
                            John Brown
                          </td>
                          <td
                            class=""
                          >
                            32
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            New York No. 1 Lake Park
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="2"
                        >
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <span
                              class="ant-table-row-indent indent-level-0"
                              style="padding-left: 0px;"
                            />
                            Joe Black
                          </td>
                          <td
                            class=""
                          >
                            42
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            London No. 1 Lake Park
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="3"
                        >
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <span
                              class="ant-table-row-indent indent-level-0"
                              style="padding-left: 0px;"
                            />
                            Jim Green
                          </td>
                          <td
                            class=""
                          >
                            32
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            Sidney No. 1 Lake Park
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="4"
                        >
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <span
                              class="ant-table-row-indent indent-level-0"
                              style="padding-left: 0px;"
                            />
                            Jim Red
                          </td>
                          <td
                            class=""
                          >
                            32
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            London No. 2 Lake Park
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination"
                unselectable="unselectable"
              >
                <li
                  aria-disabled="true"
                  class="ant-pagination-disabled ant-pagination-prev"
                  title="Previous Page"
                >
                  <a
                    class="ant-pagination-item-link"
                  >
                    <i
                      class="anticon anticon-left"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="left"
                        fill="currentColor"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </i>
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a>
                    1
                  </a>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-disabled ant-pagination-next"
                  title="Next Page"
                >
                  <a
                    class="ant-pagination-item-link"
                  >
                    <i
                      class="anticon anticon-right"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="right"
                        fill="currentColor"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                        />
                      </svg>
                    </i>
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            style="position: absolute; top: 0px; left: 0px; width: 100%;"
          >
            <div>
              <div
                class="ant-dropdown  ant-dropdown-placement-bottomRight  ant-dropdown-hidden"
              >
                <div>
                  <div
                    class="custom-filter-dropdown"
                  >
                    <input
                      class="ant-input"
                      placeholder="Search name"
                      type="text"
                      value=""
                    />
                    <button
                      class="ant-btn ant-btn-primary"
                      type="button"
                    >
                      <span>
                        Search
                      </span>
                    </button>
                    <button
                      class="ant-btn"
                      type="button"
                    >
                      <span>
                        Reset
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="position: absolute; top: 0px; left: 0px; width: 100%;"
          >
            <div>
              <div
                class="ant-dropdown  ant-dropdown-placement-bottomRight  ant-dropdown-hidden"
              >
                <div
                  class="ant-table-filter-dropdown"
                >
                  <ul
                    class="ant-dropdown-menu ant-dropdown-menu-without-submenu ant-dropdown-menu-root ant-dropdown-menu-vertical"
                    role="menu"
                    tabindex="0"
                  >
                    <li
                      class="ant-dropdown-menu-item"
                      role="menuitem"
                    >
                      <label
                        class="ant-checkbox-wrapper"
                      >
                        <span
                          class="ant-checkbox"
                        >
                          <input
                            class="ant-checkbox-input"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="ant-checkbox-inner"
                          />
                        </span>
                      </label>
                      <span>
                        London
                      </span>
                    </li>
                    <li
                      class="ant-dropdown-menu-item"
                      role="menuitem"
                    >
                      <label
                        class="ant-checkbox-wrapper"
                      >
                        <span
                          class="ant-checkbox"
                        >
                          <input
                            class="ant-checkbox-input"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="ant-checkbox-inner"
                          />
                        </span>
                      </label>
                      <span>
                        New York
                      </span>
                    </li>
                  </ul>
                  <div
                    class="ant-table-filter-dropdown-btns"
                  >
                    <a
                      class="ant-table-filter-dropdown-link confirm"
                    >
                      OK
                    </a>
                    <a
                      class="ant-table-filter-dropdown-link clear"
                    >
                      Reset
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
